src/common/transformations/include/transformations/common_optimizations/mark_subgraphs_to_keep_in_mixed_precision.hpp

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/pass/graph_rewrite.hpp"
+#include "openvino/pass/pass.hpp"
+#include "transformations_visibility.hpp"
+
+namespace ov {
+namespace pass {
+
+class TRANSFORMATIONS_API MarkSugraphsToKeepInMixedPrecision;
+
+}  // namespace pass
+}  // namespace ov
+
+/*
+ * @ingroup ie_transformation_common_api
+ * @brief: MarkSugraphsToKeepInMixedPrecision container for marking passes which marks subgraphs
+ * to be kept in f32 for mixed precision inference. Includes passes for the following patterns:
```

Comment:
Potentially yes, there can be whole `f64`, or `f16` subgraphs but 
1. in practice in 99.9% scenarios there will be `f32`, especially when the model is loaded from IR all constants will be f32. When we've run validation on the models on which this transformations are triggered we didn't face any regression because of type inconsistency error
2. to be 100% sure we also have on GPU conversion of all `f64`->`f32` in the very beginning https://github.com/openvinotoolkit/openvino/pull/14965/files#diff-57d03639eb094b00df049c347f7ca519a55138d0f7290e0aa2d67c03950bbd13R144 since GPU any wayconverts all `f32` into `f32` or `f16`, depending of inference precision but newer keeps `f64`